include/fastdds/rtps/writer/ReaderProxy.h

```diff
@@ -229,15 +188,19 @@ class ReaderProxy
 
     /**
      * Turns all UNDERWAY changes into UNACKNOWLEDGED.
+     *
      * @return true if at least one change changed its status, false otherwise.
      */
     bool perform_nack_supression();
 
     /**
      * Turns all REQUESTED changes into UNSENT.
+     *
+     * @param func Function executed for each change which changes its status.
      * @return the number of changes that changed its status.
      */
-    uint32_t perform_acknack_response();
+    uint32_t perform_acknack_response(
+            std::function<void(ChangeForReader_t& change)> func);
```

Comment:
Passing in an `std::function` by value produces dynamic allocations. You could pass a const reference, or make this a template method

Reply:
Done